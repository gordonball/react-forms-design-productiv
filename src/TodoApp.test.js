import { render } from "@testing-library/react";
import TodoApp from "./TodoApp.js";

const todo1 = {
  id: 1,
  title: "Test1",
  description: "Test2",
  priority: 1
}

const todo2 = {
  id: 2,
  title: "Test2",
  description: "Test2",
  priority: 2
}

const todo3 = {
  id: 3,
  title: "Test3",
  description: "Test3",
  priority: 3
}

describe("TodoApp", function() {
  it("renders without crashing", function() {
    render(<TodoApp initialTodos={[todo1, todo2, todo3]} />);
  });

  it("matches snapshot", function() {
    const { container } = render(
      <TodoApp initialTodos={[todo1, todo2, todo3]} />
    );

    expect(container).toMatchSnapshot();
  });
});
