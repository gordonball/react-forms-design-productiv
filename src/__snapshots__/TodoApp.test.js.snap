// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoApp matches snapshot 1`] = `
<div>
  <main
    class="TodoApp"
  >
    <div
      class="row"
    >
      <div
        class="col-md-6"
      >
        <div
          class="EditableTodoList"
        >
          <div
            class="EditableTodo"
          >
            <div
              class="mb-3"
            >
              <div
                class="float-end text-sm-end"
              >
                <button
                  class="EditableTodo-toggle btn-link btn btn-sm"
                >
                  Edit
                </button>
                <button
                  class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
                >
                  Del
                </button>
              </div>
              <div
                class="Todo"
              >
                <div>
                  <b>
                    Test1
                  </b>
                   
                  <small>
                    (priority: 
                    1
                    )
                  </small>
                </div>
                <div>
                  <small>
                    Test2
                  </small>
                </div>
              </div>
            </div>
          </div>
          <div
            class="EditableTodo"
          >
            <div
              class="mb-3"
            >
              <div
                class="float-end text-sm-end"
              >
                <button
                  class="EditableTodo-toggle btn-link btn btn-sm"
                >
                  Edit
                </button>
                <button
                  class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
                >
                  Del
                </button>
              </div>
              <div
                class="Todo"
              >
                <div>
                  <b>
                    Test2
                  </b>
                   
                  <small>
                    (priority: 
                    2
                    )
                  </small>
                </div>
                <div>
                  <small>
                    Test2
                  </small>
                </div>
              </div>
            </div>
          </div>
          <div
            class="EditableTodo"
          >
            <div
              class="mb-3"
            >
              <div
                class="float-end text-sm-end"
              >
                <button
                  class="EditableTodo-toggle btn-link btn btn-sm"
                >
                  Edit
                </button>
                <button
                  class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
                >
                  Del
                </button>
              </div>
              <div
                class="Todo"
              >
                <div>
                  <b>
                    Test3
                  </b>
                   
                  <small>
                    (priority: 
                    3
                    )
                  </small>
                </div>
                <div>
                  <small>
                    Test3
                  </small>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col-md-6"
      >
        <section
          class="mb-4"
        >
          <h3>
            Top Todo
          </h3>
          <div
            class="Todo"
          >
            <div>
              <b>
                Test1
              </b>
               
              <small>
                (priority: 
                1
                )
              </small>
            </div>
            <div>
              <small>
                Test2
              </small>
            </div>
          </div>
        </section>
        <section>
          <h3
            class="mb-3"
          >
            Add Nü
          </h3>
          <form
            class="NewTodoForm"
          >
            <div
              class="mb-3"
            >
              <label
                for="newTodo-title"
              >
                Title
              </label>
              <input
                aria-label="Title"
                class="form-control"
                id="newTodo-title"
                name="title"
                placeholder="Title"
                value=""
              />
            </div>
            <div
              class="mb-3"
            >
              <label
                for="newTodo-description"
              >
                Description
              </label>
              <textarea
                aria-label="Description"
                class="form-control"
                id="newTodo-description"
                name="description"
                placeholder="Description"
              />
            </div>
            <div
              class="mb-3 d-flex justify-content-between"
            >
              <div
                class="w-75 d-flex justify-content-between"
              >
                <label
                  class="d-inline-flex"
                  for="newTodo-priority"
                >
                  Priority:  
                </label>
                <select
                  class="form-control form-control-sm d-inline-flex"
                  id="newTodo-priority"
                  name="priority"
                >
                  <option
                    value="1"
                  >
                    Ultra-Über
                  </option>
                  <option
                    value="2"
                  >
                    Über
                  </option>
                  <option
                    value="3"
                  >
                    Meh
                  </option>
                </select>
              </div>
              <button
                class="btn-primary rig btn btn-sm NewTodoForm-addBtn"
              >
                Gø!
              </button>
            </div>
          </form>
        </section>
      </div>
    </div>
  </main>
</div>
`;
